models/chat.js

```diff
@@ -0,0 +1,19 @@
+const { Schema, model } = require('mongoose');
+const Message = require('./message.js');
+module.exports = model("Chat", new Schema({
```

Comment:
can you explain a little more why we have `users` on the room? If there's a reference to the User through the message, what's the reason for the users set? Also I can't recall the difference between 
```
users: [{ type: Schema.Types.ObjectId, ref: 'User', required: true }], <-- this type of data
messages: { type: [Message], required: true } <-- and this type of data
```

Reply:
i have a users on the room so that I can populate it with user information when retrieving data from the database. i felt it would be a little redundant to populate it on EVERY SINGLE message in the messages array. the only reason the messages have user ids attached to them is to know who sent and who recieved. i figured when loading the info on the front end you can check to see does the messages.from === user.id and if so render accordingly.

Reply:
on users, when I had the entire thing wrapped in [] it would populate correctly. if i only put the [] around the type the population would return an empty array. no clue why it works that way but you know the golden rule: if it ain't broke, don't fix it.